### DISCUSSION TITLE
Stack in case of large no. of concurrent connections

### DISCUSSION THREAD

**Discussion by shivshankardayal** [matt-42/lithium#57] [Q&A] [answered]
As far as I understand Lithium is based on co-routines so how does stack behave if there is large no. of concurrent connections. It is assumed that each co-routines allocate some data on stack.

**Comment by matt-42** [comment, marked as answer]
Hi @shivshankardayal ,

There is currently no check on the number of connections. I have to add a check to avoid crashing the server is your spam it with new connections requests. However they won't be any stack overflow in case of too many connections since the context on each connection is allocated on the heap. So you can a heap overflow yes.

**Reply by shivshankardayal** [reply]
Heap overflow is oom. I assume that system has enough RAM to handle the connections. Even though context of connection is allocated on heap the user code can allocate on stack which may lead to stack overflow. This is always a problem with coroutines when dealing with high concurrency.
